neuromation/cli/job.py

```diff
@@ -1127,3 +1169,48 @@ def calc_statuses(status: Sequence[str], all: bool) -> Set[JobStatus]:
                 return parse_columns(format_str)
         return COLUMNS
     return format
+
+
+async def calc_life_span(client: Client, value: Optional[str]) -> Optional[float]:
+    delta = (
+        _parse_timedelta(value)
+        if value is not None
+        else await calc_default_life_span(client)
+    )
+    seconds = delta.total_seconds()
+    if seconds == 0:
+        return None
+    assert seconds > 0, f"life-span cannot be negative, got: {seconds}"
+    return seconds
+
+
+async def calc_default_life_span(client: Client) -> timedelta:
+    config = await client.config.get_user_config()
+    section = config.get("job")
+    life_span = DEFAULT_JOB_LIFE_SPAN
+    if section is not None:
+        value = section.get("life-span")
+        if value is not None:
+            life_span = value
+    return _parse_timedelta(life_span)
+
+
+def _parse_timedelta(value: str) -> timedelta:
+    value = value.strip()
+    err = f"Could not parse job timeout '{value}'"
+    if value == "":
+        raise click.UsageError(f"{err}: Empty string not allowed")
+    if value == "0":
+        return timedelta(0)
+    match = REGEX_JOB_LIFE_SPAN.search(value)
+    if match is None:
+        raise click.UsageError(
+            f"{err}: Should be like '1d 2h 3m 4s' with or without spaces, "
+            "some parts may be missing"
+        )
+    return timedelta(
+        days=int(match.group("d") or 0),
+        hours=int(match.group("h") or 0),
```

Comment:
Is "1d35h120m" valid?

Reply:
yes, I think it should be valid

Reply:
The order matters, so `1h1d` will be invalid. But I think it's OK